sparkles/core.py

```diff
@@ -1038,22 +1038,37 @@ def check_guide_count(self):
                 'caution',
                 f'{obs_type} with more than {bright_cnt_lim} stars brighter than 6.1.')
 
+        # Requested slots for guide stars and mon windows
+        n_guide_or_mon_request = self.call_args['n_guide']
+
+        # Actual guide stars
         n_guide = len(self.guides)
 
-        # Different number of guide stars than requested
-        if n_guide != self.n_guide:
-            self.add_message(
-                'caution',
-                f'{obs_type} with {n_guide} guides but {self.n_guide} were requested')
+        # Actual mon windows. For catalogs from pickles from proseco < 5.0
+        # self.mons might be initialized to a NoneType or not be an attribute so
```

Comment:
When loading an pre-proseco 5.0 pickle in an environment with proseco > 5, IIRC I only got a None and not the case of not an attribute (so I assume that the new proseco in the path means that the loaded-pickle objects have the modern mon attributes?).  But the code seems fine.